src/lib.rs

```diff
@@ -8,163 +8,154 @@
  * except according to those terms.
  */
 
-extern crate cargo;
+extern crate cargo_metadata;
 extern crate time;
+extern crate toml;
 #[macro_use]
 extern crate structopt;
+#[macro_use]
+extern crate failure;
 
-use cargo::core::registry::PackageRegistry;
-use cargo::core::resolver::Method;
-use cargo::core::{Package, PackageSet, Resolve, Workspace};
-use cargo::ops;
-use cargo::util::{important_paths, CargoResult, CargoResultExt};
-use cargo::{CliResult, Config};
+pub use failure::Error; // re-exported to main
```

Comment:
Why? Main can't just pull in its own version.

Reply:
You are asking why I'm not declaring it also in the main? Because in main I'm not using logging macro if so I will update it.

Reply:
No asking why you're reexporting this which makes it part of the ABI.